website/docs/how-tos/how-to-sign-a-tx.mdx

```diff
@@ -0,0 +1,251 @@
+
+## P2PKH Signing
+
+### Required Arguments
+
+- **Private Key**: a secp256k1 private key
+- **Witnesses**: contain transaction signatures
+
+### Pseudo Code
+
+```
+def sign_tx(pk, tx):
+		# Group transaction inputs for signing
+    input_groups = compute_input_groups(tx.inputs)
+
+    # Iterate through each group of inputs
+    for indexes in input_groups:
+        group_index = indexes[0]
+
+        # Create a placeholder for the lock field in the first witness of the group
+        dummy_lock = [0] * 65 # Placeholder, assuming a 65-byte lock field
+
+        # Retrieve and deserialize the first witness in the group
+        witness = tx.witnesses[group_index]
+        witness_args = witness.deserialize()
+
+        # Replace the lock field with the dummy placeholder
+        witness_args.lock = dummy_lock
+
+        # Initialize a new BLAKE2b hasher for creating the signature hash
+        hasher = new_blake2b()
+
+        # Hash the transaction hash
+        hasher.update(tx.hash())
+
+        # Hash the first witness
+        witness_len_bytes = len(serialize(witness_args)).to_le()
+        assert(len(witness_len_bytes), 8)
+
+        # Hash the length of witness
+        hasher.update(witness_len_bytes)
+        hasher.update(serialize(witness_args))
+
+        # Hash the remaining witnesses in the group
+        for i in indexes[1:]:
+            witness = tx.witnesses[i]
+            witness_len_bytes = len(witness).to_le()
+            assert(len(witness_len_bytes), 8)
+            hasher.update(witness_len_bytes)
+            hasher.update(witness)
+
+        # Hash additional witnesses not in any input group
+        for witness in tx.witnesses[len(tx.inputs):]
+            witness_len_bytes = len(witness).to_le()
+            assert(len(witness_len_bytes), 8)
+            hasher.update(witness_len_bytes)
+            hasher.update(witness)
+
+        # Finalize the hasher to get the signature hash
+        sig_hash = hasher.finalize()
+
+        # Sign the transaction with private key
+        signature = pk.sign(sig_hash)
+
+        # Replace the dummy lock field with the actual signature in the first witness
+        witness_args.lock = signature
+
+        # Serialize the updated witness_args and update the transaction's witnesses list
+        tx.witnesses[group_index] = serialize(witness_args)
+```
+
+## Multisig Signing
+
+Much like P2PKH transactions, multisig signing is about providing signatures in witnesses to unlock the input. Plus, with each input group isolated, a single transaction can combine both multisig and P2PKH signing methods.
+
+### Required Arguments
+
+- **Private Keys**: a set of secp256k1 private keys
+- **Witnesses**: contain transaction signatures
+- **Multisig Script**: a Script defining the multisig constraint
+
+### Pseudo Code
+
+```
+def sign_tx(pks, tx, multisig_script):
+		# Group the transaction inputs for signing
+    input_groups = compute_input_groups(tx.inputs)
```

Comment:
```suggestion

```python
def sign_tx(pks, tx, multisig_script):
    # Group the transaction inputs for signing
    input_groups = compute_input_groups(tx.inputs)
```